corehq/apps/data_interfaces/models.py

```diff
@@ -1162,23 +1162,34 @@ def _handle_case_duplicate_new(self, case, rule, process_updates=False):
             existing_duplicate = None
 
         current_hash = CaseDuplicateNew.case_and_action_to_hash(case, self)
+        case_became_closed = not self.include_closed and case.closed
 
-        # Check if the new parameters have changed.
-        if existing_duplicate and existing_duplicate.hash == current_hash:
+        if not self._case_was_modified(existing_duplicate, case_became_closed, current_hash):
             # Nothing has changed. We can stop processing here
             return CaseRuleActionResult(num_updates=0)
 
+        duplicate_ids = []
         with transaction.atomic():
             if existing_duplicate:
                 existing_duplicate.delete()
-            duplicate_ids = self._create_duplicates(case, rule, current_hash)
+
+            if not case_became_closed:
+                duplicate_ids = self._create_duplicates(case, rule, current_hash)
 
         if process_updates:
             num_updates = self._update_duplicates(duplicate_ids, case, rule)
             return CaseRuleActionResult(num_updates=num_updates)
         else:
             return CaseRuleActionResult(num_updates=0)
 
+    def _case_was_modified(self, existing_duplicate, case_became_closed, current_hash):
+        if case_became_closed:
+            return True
+
+        no_property_changes = existing_duplicate and existing_duplicate.hash == current_hash
+
+        return not no_property_changes
```

Comment:
I think the intermediate variable adds context so that the reader doesn't have to realize that identical hashes implies no changes. I think its easier to understand 'not no_property_changes' than reason out the longer expression